internal/pkg/cli/cli.go

```diff
@@ -139,3 +126,26 @@ func indentListItem(multiline string) string {
 	}
 	return strings.Join(prefixedLines, "\n")
 }
+
+// CwdPathDisplayer displays paths to the user relative to their current working directory.
+type CwdPathDisplayer struct {
+	workingDir string
+}
+
+func NewCwdPathDisplayer() (*CwdPathDisplayer, error) {
```

Comment:
The principal concern here is that this would mean that test cases still end up calling `filepath.Rel`. The root problem is that there exist input strings to `Rel` on Windows that cannot possibly be relativized to *any* given directory on Windows even though they might work on Linux.

I saw that you had already suggested this approach but because of the issues in the previous paragraph, it is not a simple solution. It doesn't matter if we can change the working directory that gets passed to `relPath` because no matter what the value of `cwdPath` might be, when `relPath` is called with `relPath(cwdPath, "/some/bogus/test/path")` that will *always* result in an error on Windows because the second path is not a valid path on the host filesystem. On Linux and Mac, of course, path rules are super lenient because `/` is the root and the separator, and every other character than `/` is a valid file name character, so these test cases do not cause problems on Linux and Mac. So, if I were to go forward with this approach, I would have to modify a large number of path strings in a few large test files, which would be a substantial amount of work.

But when I look at the details of these test cases, it's clear that for the most part, they are already mostly mocked away from touching the filesystem. So the fact that filepath.Rel was still getting called feels wrong - since these paths and filesystem accesses that are seemingly supposed to be mocked away are leaking through to the host filesystem - and that doesn't seem quite right to me.

This was why I used the `pathDisplayer` interface approach. Since the only current need for making a path relative to the current working directory is a detail of displaying paths to users, this felt like a better point of extensibility because it means that in test cases we can operate on file paths that are invalid on the current host file system, because we are able to know that the only significance of the result of calling DisplayPath is that it will be printed to the console - not that it will then be used to access any files on the file system. It makes it obvious what the purpose of computing the relative path is.

If you are still confident that I should remove the pathDisplayer interface I am willing to do so, but this is why I think it's a reasonable approach to the problem caused by bogus file paths in test cases.

Reply:
Thanks for the explanation Ryan!
I'm still not inclined to using a `pathDisplayer` interface and field as is because reading `o.pathDisplayer.DisplayPath(summary.Path)` isn't clear to me what i'm expected to get. 

What if we allowed overriding `filepath.Rel` by embedding it as a field in our types?
```go
type initEnvOpts struct {
    cwd       string
    relPath   func(string, string) (string, error)  // Overriden in tests to work with Windows filepaths.
}

func newInitEnvOpts(vars initEnvVars) (*initEnvOpts, error) { {
   wd, err := os.Getwd()
   if err != nil {
      return nil, err
   }
   return &initEnvOpts{
       // assign other fields...
       cwd:     wd,
       relPath: relPath,
   }
}
```

Reply:
That makes sense. I will say the part about it not being fully clear about the expected result is partially by design - since the function is meant to just return a path that is appropriate to display to the user in the CLI. Though, that might be Java-ish approach carrying over - is it preferred in Go to avoid these kinds of general abstractions?

That proposed approach would work, but the main concern with it is that it would make the test cases look something like this:

```go
opts.cwd = "/test/"
opts.relPath = func(base string, fullPath string) (string, error) {
    if base == "/test/" && fullPath == "/test/Dockerfile" {
         return "Dockerfile";
    } else {
        panic("Unexpected relPath call: (" + base + ", " + fullPath + ")")
    }
}
```

Which seems a whole lot less nice to look at than:

```go
mockPathDisplayer = mocks.newMockPathDisplayer(ctrl)
opts.pathDisplayer = mockPathDisplayer
mockPathDisplayer.EXPECT().DisplayPath("/test/Dockerfile").Return("Dockerfile", nil)
```

Especially since the first two lines of the second case are shared with all test cases, while with the previous approach that function has to be repeated for every test case in the table.

Reply:
>is it preferred in Go to avoid these kinds of general abstractions?

It doesn't have anything to do with the language, my concern is that we are introducing an abstraction to make testing on Windows possible. Since DisplayPath is just for pretty-printing information and isn't critical what if we just define the error out of existence?

```go
func displayPath(base, target string) string {
	if !filepath.IsAbs(target) {
		return filepath.Clean(target)
	}

	rel, err := filepath.Rel(base, target)
	if err != nil {
		// No path from base to target available, return target as is.
		return filepath.Clean(target)
	}
	return rel
}
```
This way we don't have to touch the tests either

Reply:
Okay, that approach makes sense, I will implement it! Thanks for all the feedback 👍